app/models/storage.rb

```diff
@@ -832,4 +832,33 @@ def perf_capture(interval_name)
   def vm_scan_affinity
     with_relationship_type("vm_scan_storage_affinity") { parents }
   end
+
+  # is_available?
+  # Returns:  true or false
+  #
+  # The UI calls this method to determine if a feature is supported for this Storage
+  # and determines if a button should be displayed.  This method should return true
+  # even if a function is not 'currently' available due to some condition that is not
+  # being met.
+  def is_available?(request_type)
+    send("validate_#{request_type}")[:available]
+  end
+
+  # is_available_now_error_message
+  # Returns an error message string if there is an error.
+  # Returns nil to indicate no errors.
+  # This method is used by the UI along with the is_available? methods.
+  def is_available_now_error_message(request_type)
+    send("validate_#{request_type}")[:message]
+  end
+
+  def self.batch_operation_supported?(operation, ids)
+    Storage.where(:id => ids).all? { |s| s.public_send("validate_#{operation}")[:available] }
+  end
+
+  def validate_smartstate_analysis
+    return {:available => false, :message => "Smartstate Analysis cannot be performed on selected Datastore"} if ext_management_systems.blank? ||
+                                                     !ext_management_systems.first.kind_of?(ManageIQ::Providers::Vmware::InfraManager)
```

Comment:
My other comment on this line got buried, so I'll repeat.  Why are we only checking the `.first` one?  If it's attached to multiple EMSes (not sure that's possible), then it could randomly flip.